query/call_iterator.go

```diff
@@ -39,6 +39,11 @@ func NewCallIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	name := opt.Expr.(*influxql.Call).Name
 	switch name {
 	case "count":
+		// Count is optimized in the lower level storage to return an aggregate sum,
+		// per window, if there are no field conditions.
+		if opt.IndexOnlyConditions() {
```

Comment:
The issue I ran into (and why I did it this way) was that a `count` seems to expects N values to be returned from the series and it counts each one.  The fast iterator return 1 point with the value as the count.  Using the existing logic, it would see that as 1 value.  I needed it to sum each point though.

I'll need to pair up with you so I can better understand what you mean.